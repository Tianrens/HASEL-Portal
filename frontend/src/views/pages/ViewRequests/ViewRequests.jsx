import { React } from 'react';
import { TableRow, TableCell } from '@material-ui/core';
import { Link } from 'react-router-dom';
import { container } from './ViewRequests.module.scss';
import TopBarPageTemplate from '../../components/templates/TopBarPageTemplate/TopBarPageTemplate';
import StyledHeader from '../../components/text/StyledHeader';
import TableWithPagination from '../../components/table/TableWithPagination';
import { getDisplayName } from '../../../config/accountTypes';

const columns = ['Name', 'Account Type', 'Application Received'];

const rowFactory = (row) => (
    <TableRow key={row._id} component={Link} to={`/requests/${row._id}`}>
        <TableCell component='th' scope='row'>
            {row.userId.firstName} {row.userId.lastName}
        </TableCell>
        <TableCell align='right'>{getDisplayName(row.userId.type)}</TableCell>
        <TableCell align='right'>{new Date(row.createdAt).toLocaleDateString()}</TableCell>
    </TableRow>
);

const sections = [
    { title: 'Pending Requests', endpoint: 'PENDING' },
    { title: 'Approved Requests', endpoint: 'ACTIVE' },
    { title: 'Denied Requests', endpoint: 'DECLINED' },
];

const ViewRequests = () => (
    <TopBarPageTemplate>
        <div className={container}>
            {sections.map((section) => (
                <div key={section.title}>
                    <StyledHeader left>{section.title}</StyledHeader>
                    <TableWithPagination
                        endpoint={`/api/request/status/${section.endpoint}`}
                        rowProp='requests'
                        columns={columns}
                        rowFactory={rowFactory}
                    />
                </div>
            ))}
        </div>
    </TopBarPageTemplate>
);

export default ViewRequests;
