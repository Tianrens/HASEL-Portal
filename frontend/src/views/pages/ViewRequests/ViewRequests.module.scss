@import '../../../assets/colours';
@import '../../../assets/responsiveDesignHelpers';

.container {
    display: flex;
    flex-direction: column;
    width: 70%;
    height: 100%;
    margin: auto;
}

// Make name inputs appear in a column
@include responsive(mobile) {
    .container {
        width: 100%;
    }
}
