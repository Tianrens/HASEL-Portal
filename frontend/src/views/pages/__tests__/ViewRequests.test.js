import React from 'react';
import renderer from 'react-test-renderer';
import StylesProvider from '@mui/styles/StylesProvider';
import { MemoryRouter } from 'react-router-dom';
import ViewRequests from '../ViewRequests/ViewRequests';

test('View Requests page renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <MemoryRouter>
                <ViewRequests />
            </MemoryRouter>
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
