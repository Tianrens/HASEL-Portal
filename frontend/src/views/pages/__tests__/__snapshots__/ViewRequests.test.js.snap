// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Requests page renders properly 1`] = `
Array [
  <div
    className="fillTopbarSpace"
  >
    <div
      className="navbar"
    >
      <div
        className="patternBgLayer"
      >
        <div
          className="bgLayer"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="University of Auckland"
              className="uoaLogo"
              src="uoaLogo.svg"
            />
            <div
              className="spacer"
            />
            <div
              className="title"
            >
              HASEL Lab
            </div>
          </a>
          <div
            className="pushSpacer"
          />
          <div
            className="horizontalNavLinks"
          >
            <div
              className="linksWrapper"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text wrapper"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <a
                    href="/user"
                    onClick={[Function]}
                  >
                    <img
                      alt="My Account"
                      className="navIcon"
                      src="id.svg"
                    />
                    <div
                      className="spacer"
                    />
                    <div
                      className="title"
                    >
                      My Account
                    </div>
                  </a>
                </span>
              </button>
            </div>
          </div>
          <div
            className="drawerWrapper"
            onClick={[Function]}
            onTouchEnd={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root drawerButton"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M2 15.5v2h20v-2H2zm0-5v2h20v-2H2zm0-5v2h20v-2H2z"
                  />
                </svg>
              </span>
            </button>
            <div
              className="MuiDrawer-root MuiDrawer-docked drawer"
            >
              <div
                className="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
                style={
                  Object {
                    "visibility": "hidden",
                  }
                }
              >
                <div
                  className="drawerPatternBg"
                >
                  <div
                    className="drawer"
                  >
                    <div>
                      <button
                        className="MuiButtonBase-root MuiIconButton-root drawerButton"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                    <div
                      className="linksWrapper"
                    >
                      <button
                        className="MuiButtonBase-root MuiButton-root MuiButton-text wrapper"
                        disabled={false}
                        onBlur={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiButton-label"
                        >
                          <a
                            href="/user"
                            onClick={[Function]}
                          >
                            <img
                              alt="My Account"
                              className="navIcon"
                              src="id.svg"
                            />
                            <div
                              className="spacer"
                            />
                            <div
                              className="title"
                            >
                              My Account
                            </div>
                          </a>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="content"
  >
    <div
      className="container"
    >
      <div>
        <div
          className="container"
        >
          <h2
            className="text alignLeft "
          >
            <span>
              Pending Requests
            </span>
          </h2>
        </div>
        <div
          className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <table
            className="MuiTable-root table"
            role={null}
          >
            <thead
              className="MuiTableHead-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head headerCell"
                  scope="col"
                >
                  Name
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head headerCell"
                  scope="col"
                >
                  Account Type
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head headerCell"
                  scope="col"
                >
                  Application Received
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root"
              role={null}
            />
          </table>
        </div>
        <div
          className="MuiTablePagination-root"
        >
          <div
            className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
          >
            <div
              className="MuiTablePagination-spacer"
            />
            <p
              className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            >
              0-0 of 0
            </p>
            <div
              className="MuiTablePagination-actions"
            >
              <button
                aria-label="Previous page"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Previous page"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="Next page"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Next page"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="container"
        >
          <h2
            className="text alignLeft "
          >
            <span>
              Approved Requests
            </span>
          </h2>
        </div>
        <div
          className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <table
            className="MuiTable-root table"
            role={null}
          >
            <thead
              className="MuiTableHead-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head headerCell"
                  scope="col"
                >
                  Name
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head headerCell"
                  scope="col"
                >
                  Account Type
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head headerCell"
                  scope="col"
                >
                  Application Received
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root"
              role={null}
            />
          </table>
        </div>
        <div
          className="MuiTablePagination-root"
        >
          <div
            className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
          >
            <div
              className="MuiTablePagination-spacer"
            />
            <p
              className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            >
              0-0 of 0
            </p>
            <div
              className="MuiTablePagination-actions"
            >
              <button
                aria-label="Previous page"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Previous page"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="Next page"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Next page"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="container"
        >
          <h2
            className="text alignLeft "
          >
            <span>
              Denied Requests
            </span>
          </h2>
        </div>
        <div
          className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <table
            className="MuiTable-root table"
            role={null}
          >
            <thead
              className="MuiTableHead-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head headerCell"
                  scope="col"
                >
                  Name
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head headerCell"
                  scope="col"
                >
                  Account Type
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head headerCell"
                  scope="col"
                >
                  Application Received
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root"
              role={null}
            />
          </table>
        </div>
        <div
          className="MuiTablePagination-root"
        >
          <div
            className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
          >
            <div
              className="MuiTablePagination-spacer"
            />
            <p
              className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            >
              0-0 of 0
            </p>
            <div
              className="MuiTablePagination-actions"
            >
              <button
                aria-label="Previous page"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Previous page"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="Next page"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                title="Next page"
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
